project.yaml

```diff
@@ -1733,12 +1735,23 @@ Layer:
     <<: *extents
     Datasource:
       <<: *osm2pgsql
-      table: |2-
-              (SELECT way,"addr:housenumber", way_area FROM planet_osm_polygon WHERE "addr:housenumber" IS NOT NULL AND building IS NOT NULL
-               UNION
-               SELECT way,"addr:housenumber", NULL AS way_area FROM planet_osm_point WHERE "addr:housenumber" IS NOT NULL
-               ORDER BY way_area DESC
-              ) AS housenumbers
+      table: |-
+        (SELECT
+            way,
+            "addr:housenumber",
+            way_area/NULLIF(!pixel_width!::real*!pixel_height!::real,0) AS way_pixels
+          FROM planet_osm_polygon
+          WHERE "addr:housenumber" IS NOT NULL
+            AND building IS NOT NULL
```

Comment:
Unless it's been added since we switched to carto, I don't think so. I've recommended adding a `building IS NULL` index, which is the inverse.